packages/trader/src/Modules/Trading/Components/Elements/__tests__/purchase-button.spec.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { isDesktop, isMobile } from '@deriv/shared';
+import PurchaseButton from '../purchase-button';
+
+const default_mocked_props = {
+    basis: '',
+    buy_info: { error: '' },
+    currency: '',
+    growth_rate: 0.03,
+    has_deal_cancellation: false,
+    index: 1,
+    info: {
+        has_error: false,
+        id: 'test_id',
+        message: 'test_message',
+        growth_rate: 0.03,
+        stake: '10',
+    },
+    is_accumulator: false,
+    is_disabled: false,
+    is_high_low: false,
+    is_loading: false,
+    is_multiplier: false,
+    is_proposal_empty: false,
+    is_vanilla: false,
+    is_turbos: false,
+    onClickPurchase: jest.fn(),
+    purchased_states_arr: [true, false],
+    should_fade: false,
+    setPurchaseState: jest.fn(),
+    type: 'VANILLALONGCALL',
+};
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(() => false),
+    isDesktop: jest.fn(() => true),
+}));
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    IconTradeTypes: jest.fn(props => <p data-type={props.type}>TradeIcon</p>),
+    Money: jest.fn(() => <p>MoneyComponent</p>),
```

Comment:
@kate-deriv is it necessary to mock Money and check if MoneyComponent is in the document? can we not mock it and check the actual sum of money that we pass as info.stake: getByText('10 USD')? we'll need to pass currency as 'USD' for this to work  :)

Reply:
I tried to mock most of the non-wrapper-components used as children in order to isolate testing of the parent (main) component. I read about this approach (so called 'London school') in 'Unit Testing Principles, Practices, and Patterns' and decided to enrich my practice of writing tests. This way of writing tests has a lot of benefits, like: easy to maintain, no need to refactor after each change of one of the child components and etc. Even if the tests will fail, you'll always know in which component the problem is (no need in research between children components). After all it 's unit, not integration test.

<img width="669" alt="Screenshot 2023-09-19 at 9 12 22 AM" src="https://github.com/binary-com/deriv-app/assets/121025168/44a7e335-e955-4a00-8591-3a2dab92de39">
<img width="669" alt="Screenshot 2023-09-19 at 9 13 44 AM" src="https://github.com/binary-com/deriv-app/assets/121025168/922bcddd-0441-4013-9c8b-5d64a4b19b5d">
<img width="669" alt="Screenshot 2023-09-19 at 9 13 54 AM" src="https://github.com/binary-com/deriv-app/assets/121025168/fce75d13-e743-4dd7-8e34-fd1584d7ce99">

Reply:
thank you for explanation @kate-deriv :) i didn't know you wanted to avoid integration testing here